### PR TITLE
Add .pre-commit-config.yaml

### PR DESCRIPTION
**Pull request by maestroque**
Add a pre-commit configuration file, as in [phys2bids](https://github.com/physiopy/phys2bids)

## Proposed Changes
- All the python style checkers are added
- rst checkers are added as is from phys2bids

## Change Type

- [ ] `bugfix` (+0.0.1)
- [ ] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [ ] `refactoring` (no version update)
- [ ] `test` (no version update)
- [x] `infrastructure` (no version update)
- [ ] `documentation` (no version update)
- [ ] `other`

## Checklist before review

- [ ] I added everything I wanted to add to this PR.
- [ ] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [ ] \[Code or tests only\] I ran and passed tests locally.
- [ ] \[Documentation only\] I built the docs locally.
- [ ] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [ ] My code respects the adopted style, especially linting conventions.
- [ ] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [ ] I added or indicated the right labels.

- [ ] I added information regarding the timeline of completion for this PR.

- [ ] Please, comment on my PR while it's a draft and give me feedback on the development!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE_DISCUSSION.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 name: Discussion
-about: Use this template to start a discussion issue, i.e. an issue that 
+about: Use this template to start a discussion issue, i.e. an issue that
```
Comment (smoia) [inline .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE_DISCUSSION.md:3]:
```suggestion
about: Use this template to start a discussion issue, i.e. an issue meant to open a community debate over a topic
```

File: .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE_MEETING.md [hunk 2]
```diff
@@ -8,6 +8,6 @@ assignees: smoia
```
Comment (smoia) [inline .github/ISSUE_TEMPLATE/ISSUE_TEMPLATE_MEETING.md:8]: This file can be actually removed

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    rev: v4.5.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+    -   id: check-case-conflict
+    -   id: check-merge-conflict
+-   repo: https://github.com/psf/black
+    rev: 23.11.0
```
Comment (smoia) [inline .pre-commit-config.yaml:14]: It is worth updating to the latest version of all the hooks!

File: setup.cfg [hunk 4]
```diff
@@ -65,7 +65,7 @@ all =
```
Comment (smoia) [inline setup.cfg:65]: In the setup file you should declare the options for pre-commit and its hooks (see phys2bids setup.cfg)

**Review by smoia — CHANGES_REQUESTED** [review]
I see some formatting has happened ;)

The main feedback is to change the setup.cfg configuration - you can take a look at phys2bids' or nigsp's one to see how to set pre-commit and its hooks up!

**Comment by smoia** [comment]
This may benefit from having a look at #41 (?)

File: peakdet/operations.py [hunk 5]
```diff
@@ -139,7 +139,7 @@ def peakfind_physio(data, *, thresh=0.2, dist=None):
     heights = np.percentile(heights["peak_heights"], 1)
     locs, heights = signal.find_peaks(data[:], distance=cdist, height=heights)
     data._metadata["peaks"] = locs
-    # perform trough detection based on detected peaks
+    # perform through detection based on detected peaks
```
Comment (maestroque) [inline peakdet/operations.py:142]: hmmm

**Comment by maestroque** [comment]
@smoia @me-pic @m-miedema  I believe this is ready, feel free to have a look

**Comment by maestroque** [comment]
@me-pic @m-miedema @smoia This should be ready to merge. I opened #68 for the failing neurokit tests, since they are outdated and I suppose they will be addressed in another PR (or will be re-written completely). For the time being I just added a `# noqa`

File: setup.cfg [hunk 6]
```diff
@@ -65,19 +69,41 @@ all =
...
+match =
+    peakdet/*.py
+match_dir = peakdet/[^tests]*
+
+[codespell]
+skip = venvs,.venv,versioneer.py,.git,build,./docs/_build
+write-changes =
+count =
+quiet-level = 3
+
```
Comment (smoia) [inline setup.cfg:106]: Might be worth adding black options as well?

**Review by smoia — CHANGES_REQUESTED** [review]
Looks like the changes worked, good job!

Did you run tests locally? If not, could you do that, please? If they break on something like the old workflows, that's fine, but otherwise it's good to check that the blacking of the scripts didn't break anything (shouldn't, but just to be sure!)

File: setup.cfg (hunk 6 above)
Comment (maestroque) [reply setup.cfg:106]: There is no support for `setup.cfg` [(issue)](https://github.com/psf/black/issues/683), we would have to add a `pyproject.toml` instead.
I don't see any desired configuration option other than perhaps the `--exclude` option.

However, I don't see any real benefit in excluding files like `versioneer.py` and not having style uniformity. Let me know your thoughts.

**Comment by maestroque** [comment]
@smoia I ran them locally yes, they have the same behavior as in master branch currently.

File: setup.cfg [hunk 7]
```diff
@@ -65,19 +69,41 @@ all =
...
     peakdet/tests/data/*
 
 [flake8]
 exclude=
     *build/
-    heuristics
     tests
+    versioneer.py
 ignore = E126, E203, E402, W503
 max-line-length = 99
```
Comment (smoia) [inline setup.cfg:81]: Sorry, you are right about black in setup.cfg.

I got confused with extended Flake8 settings to match Black:

ignore = E126, E402, W503, F401, F811
max-line-length = 88
extend-ignore = E203, E501
extend-select = B950

**Review by smoia — CHANGES_REQUESTED** [review]

**Comment by smoia** [comment]
> @smoia I ran them locally yes, they have the same behavior as in master branch currently.

This is awesome though!

File: setup.cfg (hunk 7 above)
Comment (maestroque) [reply setup.cfg]: Done

**Comment by maestroque** [comment]
@smoia @m-miedema @me-pic this should be ready to go now

**Review by smoia — APPROVED** [review]
Awesome!

**Pull request merged**